src/libply/aux/perf_event.c

```diff
@@ -73,6 +73,42 @@ int perf_event_attach(struct ply_probe *pb, const char *path,
 	return fd;
 }
 
+int perf_event_attach_raw(struct ply_probe *pb, int type, unsigned long config,
+			  unsigned long long period, int task_mode)
+{
+	struct perf_event_attr attr = {};
+	int fd;
+	int pid = task_mode ? 0 : -1;
+	int cpu = task_mode ? -1 : 0;
+	int group_fd = task_mode ? -1 : pb->ply->group_fd;
+
+	attr.type = type;
+	attr.config = config;
+	attr.sample_type = PERF_SAMPLE_RAW;
+	attr.sample_period = period;
+	attr.wakeup_events = 1;
+	attr.disabled = 1;
+
+	fd = perf_event_open(&attr, pid, cpu, group_fd, 0);
+	if (fd < 0)
+		return -errno;
+
+	/* if (ioctl(fd, PERF_EVENT_IOC_ENABLE, 0)) { */
+	/* 	close(fd); */
+	/* 	return -errno; */
+	/* } */
```

Comment:
debug leftover?

Reply:
Right, it's just copied from the perf_event_attach function, but we might need to remove it as it's enabled later.